[Index Management] Add option to convert to lookup index in the manage index dropdown

damian-polewski opened the pull request:
## Summary

Closes #233996

This PR introduces the core logic and UI for converting an index to a lookup index in Index Management, including a custom React hook, modal enhancements, and integration of the reindex service.

Changes:
- Added an option to convert an index to a lookup index in the manage index dropdown.
- Aligned the `ConvertIndexToLookupModal` with latest design.
- Added a new `useConvertIndexToLookup` hook to handle the conversion logic.
- Integrated the `reindexService` with `index_management` plugin.
- Updated the `fetchIndicesCall` to return `settings.index.mode` of the index.

### Demo

https://github.com/user-attachments/assets/5cd578bf-d462-4cba-b753-b952cd987c2c

### How to test

1. Create an index that meets requirements (less than 2 billion docs, single shard).
> [!NOTE]  
> If the index does not meet the requirements, the button will be disabled. For indices that already have a lookup mode, the button will not be visible at all.
2. Navigate to index list and select the created index
3. Select `Convert to lookup index` option from the `Manage index` dropdown
4. Change the new lookup index name if needed and press `Convert`
5. After conversion is done you will be navigated to the lookup index's details page

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

## Release Note

Adds an option to convert an index to a lookup index in the manage index dropdown.

damian-polewski committed:
feat: wip

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

damian-polewski committed:
feat: use reindexService

damian-polewski committed:
fix: dont allow converting if index already is in lookup mode

damian-polewski committed:
feat: add isConverting and errorMessage props to convert modal

damian-polewski committed:
feat: cleanup logic

damian-polewski committed:
add use_convert_index_to_lookup hook

damian-polewski committed:
feat: disable cancel button when reindexing

damian-polewski committed:
feat: add unit tests for use_convert_index_to_lookup hook

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

florent-leborgne reviewed (commented):
Some small copy suggestions. Some might not work due to space constraints. Lmk what you think.

Edit: just saw your question, "Convert to lookup index" sounds good to me. I'm wondering if we should highlight somewhere what are the main differences when an index is in lookup mode.

jovana-andjelkovic reviewed (approved):
@damian-polewski lgtm!

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

damian-polewski committed:
fix: adjust tests

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

kapral18 commented on x-pack/platform/plugins/shared/index_management/public/application/hooks/use_convert_index_to_lookup/use_convert_index_to_lookup.test.tsx:
```diff
@@ -0,0 +1,122 @@
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
```
the counterpart to `useFakeTimers` seems to only be used in last test.  Should we move this there as well?

kapral18 replied:
or the contrary make fakeTimers default for all tests. Up to you.

Here is an updated test list with some edge cases taking into account my comments from source code

```tsx
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { renderHook, act } from '@testing-library/react';
import { ReindexStatus } from '@kbn/reindex-service-plugin/common';
import { useConvertIndexToLookup } from './use_convert_index_to_lookup';
import { getReindexStatus, startReindex } from '../../services';

jest.mock('../../services');

const mockedGetReindexStatus = getReindexStatus as jest.Mock;
const mockedStartReindex = startReindex as jest.Mock;

const defaultHookArgs = {
  sourceIndexName: 'my-index',
  onSuccess: jest.fn(),
  onClose: jest.fn(),
};

describe('useConvertIndexToLookup', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  describe('successful conversion', () => {
    it('should poll until completion', async () => {
      mockedStartReindex.mockResolvedValue({ data: {}, error: null });
      mockedGetReindexStatus
        .mockResolvedValueOnce({
          data: { reindexOp: { status: ReindexStatus.inProgress } },
          error: null,
        })
        .mockResolvedValueOnce({
          data: { reindexOp: { status: ReindexStatus.inProgress } },
          error: null,
        })
        .mockResolvedValueOnce({
          data: {
            reindexOp: {
              status: ReindexStatus.completed,
              newIndexName: 'lookup-my-index',
            },
          },
          error: null,
        });

      const { result } = renderHook(() => useConvertIndexToLookup(defaultHookArgs));

      act(() => {
        result.current.convert('lookup-my-index');
      });

      expect(result.current.isConverting).toBe(true);

      // First poll after startReindex
      await act(async () => {
        await jest.advanceTimersByTimeAsync(0);
      });

      // Second poll after 3s
      await act(async () => {
        await jest.advanceTimersByTimeAsync(3000);
      });

      // Third poll after another 3s - completes
      await act(async () => {
        await jest.advanceTimersByTimeAsync(3000);
      });

      expect(defaultHookArgs.onSuccess).toHaveBeenCalledWith('lookup-my-index');
      expect(defaultHookArgs.onClose).toHaveBeenCalled();
      expect(result.current.isConverting).toBe(false);
      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(3);
    });

    it('should handle completion without newIndexName', async () => {
      mockedStartReindex.mockResolvedValue({ data: {}, error: null });
      mockedGetReindexStatus.mockResolvedValue({
        data: { reindexOp: { status: ReindexStatus.completed } },
        error: null,
      });

      const { result } = renderHook(() => useConvertIndexToLookup(defaultHookArgs));

      act(() => {
        result.current.convert('lookup-my-index');
      });

      await act(async () => {
        await jest.advanceTimersByTimeAsync(0);
      });

      expect(defaultHookArgs.onSuccess).not.toHaveBeenCalled();
      expect(defaultHookArgs.onClose).toHaveBeenCalled();
    });
  });

  describe('error handling', () => {
    it('should stop polling on status error', async () => {
      mockedStartReindex.mockResolvedValue({ data: {}, error: null });
      mockedGetReindexStatus.mockResolvedValue({
        data: null,
        error: new Error('Status check failed'),
      });

      const { result } = renderHook(() => useConvertIndexToLookup(defaultHookArgs));

      act(() => {
        result.current.convert('lookup-my-index');
      });

      await act(async () => {
        await jest.advanceTimersByTimeAsync(0);
      });

      expect(result.current.errorMessage).toBe('Status check failed');
      expect(result.current.isConverting).toBe(false);

      // Verify no more polls happen
      act(() => {
        jest.advanceTimersByTime(6000);
      });

      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(1);
    });

    it('should handle failed reindex status', async () => {
      mockedStartReindex.mockResolvedValue({ data: {}, error: null });
      mockedGetReindexStatus.mockResolvedValue({
        data: { reindexOp: { status: ReindexStatus.failed } },
        error: null,
      });

      const { result } = renderHook(() => useConvertIndexToLookup(defaultHookArgs));

      act(() => {
        result.current.convert('lookup-my-index');
      });

      await act(async () => {
        await jest.advanceTimersByTimeAsync(0);
      });

      expect(result.current.isConverting).toBe(false);
      expect(defaultHookArgs.onSuccess).not.toHaveBeenCalled();
      expect(defaultHookArgs.onClose).toHaveBeenCalled();
    });
  });

  describe('cleanup behavior', () => {
    it('should clear timers on unmount during polling', async () => {
      mockedStartReindex.mockResolvedValue({ data: {}, error: null });
      mockedGetReindexStatus.mockResolvedValue({
        data: { reindexOp: { status: ReindexStatus.inProgress } },
        error: null,
      });

      const { result, unmount } = renderHook(() => useConvertIndexToLookup(defaultHookArgs));

      act(() => {
        result.current.convert('lookup-my-index');
      });

      // Start first poll
      await act(async () => {
        await jest.advanceTimersByTimeAsync(0);
      });

      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(1);

      unmount();

      // Advance past next poll time
      act(() => {
        jest.advanceTimersByTime(6000);
      });

      // No additional polls should occur
      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(1);
    });

    it('should clear timers when sourceIndexName changes', async () => {
      mockedStartReindex.mockResolvedValue({ data: {}, error: null });
      mockedGetReindexStatus.mockResolvedValue({
        data: { reindexOp: { status: ReindexStatus.inProgress } },
        error: null,
      });

      const { result, rerender } = renderHook((props) => useConvertIndexToLookup(props), {
        initialProps: defaultHookArgs,
      });

      act(() => {
        result.current.convert('lookup-my-index');
      });

      // Start polling
      await act(async () => {
        await jest.advanceTimersByTimeAsync(0);
      });

      // Change source index
      rerender({
        ...defaultHookArgs,
        sourceIndexName: 'different-index',
      });

      // Advance time - should not poll old index
      act(() => {
        jest.advanceTimersByTime(6000);
      });

      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(1);
      expect(mockedGetReindexStatus).toHaveBeenCalledWith('my-index');
    });
  });

  describe('timing verification', () => {
    it('should poll at correct intervals', async () => {
      mockedStartReindex.mockResolvedValue({ data: {}, error: null });
      mockedGetReindexStatus.mockResolvedValue({
        data: { reindexOp: { status: ReindexStatus.inProgress } },
        error: null,
      });

      const { result } = renderHook(() => useConvertIndexToLookup(defaultHookArgs));

      act(() => {
        result.current.convert('lookup-my-index');
      });

      // Initial poll
      await act(async () => {
        await jest.advanceTimersByTimeAsync(0);
      });
      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(1);

      // Poll after exact interval
      await act(async () => {
        await jest.advanceTimersByTimeAsync(3000);
      });
      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(2);

      // No poll before interval
      await act(async () => {
        await jest.advanceTimersByTimeAsync(2999);
      });
      expect(mockedGetReindexStatus).toHaveBeenCalledTimes(2);
    });
  });
});

```

damian-polewski replied:
Thanks for the feedback I adjusted unit tests in ee67fe3

kapral18 reviewed (commented):
Great job. Left some comments.

alisonelizabeth commented:
Nice job on this @damian-polewski 👏 

I have some general feedback, leaving here for now but we can discuss more broadly with the team if necessary.

1. Can we add a description to the modal, explaining what a lookup index is/why you would want to convert to it? Also, it may be helpful to call out when you "convert" the index, it is not actually replacing the existing index.
2. You can still click the "x" icon while the reindex is in progress. I think this is correct behavior, but maybe we need some indication that the reindexing progress will not be canceled/is still in progress. Maybe a toast would be a simple solution for now?
3. An index with no documents cannot be reindexed to a lookup index. Is that expected?
4. Should we allow system indices to be reindexed to a lookup index?

mattkime commented on x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/index_actions_context_menu/index_actions_context_menu.js:
```diff
@@ -20,11 +20,19 @@ import {
+  EuiText,
 } from '@elastic/eui';
 
+import { ConvertToLookupIndexModalContainer } from '../details_page/convert_to_lookup_index_modal/convert_to_lookup_index_modal_container';
```
How much trouble would it be to convert this file to typescript? There are very few js files left in our code base and they're always high priority to convert to ts. Might be best done as a separate PR.

mattkime commented on x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/index_actions_context_menu/index_actions_context_menu.js:
```diff
@@ -243,6 +251,41 @@ export class IndexActionsContextMenu extends Component {
+                <EuiText size="xs">
+                  <FormattedMessage
+                    id="xpack.idxMgmt.indexActionsMenu.convertToLookupIndexButton.error"
+                    defaultMessage="Index too large to be converted."
```
>and less than a single shard to be converted.

and a single shard to be converted.

---

If you figure out zero shard storage, some people on the ES side will be very interested. ;)

damian-polewski replied:
<img width="331" height="146" alt="Screenshot 2025-09-16 at 11 32 11" src="https://github.com/user-attachments/assets/639dc597-4360-43f4-8f79-86a1ce433357" />

Updated in 060ae03

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

damian-polewski committed:
fix: update copy part1

damian-polewski commented on x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/index_actions_context_menu/index_actions_context_menu.js:
```diff
@@ -462,6 +505,60 @@ export class IndexActionsContextMenu extends Component {
+
+          notificationService.showSuccessToast(
+            i18n.translate('xpack.idxMgmt.convertToLookupIndexAction.indexConvertedToastTitle', {
+              defaultMessage: 'Lookup index created',
```
<img width="359" height="158" alt="Screenshot 2025-09-16 at 11 25 05" src="https://github.com/user-attachments/assets/47a61f2b-c914-40cb-b25f-5a78e91e4d05" />
Updated in [7f1f7f0](https://github.com/elastic/kibana/pull/233998/commits/7f1f7f0ee1d1a5e2f99a3a9181c85466d73f3042)

damian-polewski commented on x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/index_actions_context_menu/index_actions_context_menu.js:
```diff
@@ -462,6 +505,60 @@ export class IndexActionsContextMenu extends Component {
+              defaultMessage: 'Lookup index created',
+            }),
+            i18n.translate('xpack.idxMgmt.convertToLookupIndexAction.indexConvertedToastMessage', {
+              defaultMessage: 'The {lookupIndexName} has been created.',
```
<img width="359" height="158" alt="Screenshot 2025-09-16 at 11 25 05" src="https://github.com/user-attachments/assets/c9285f2a-3a15-43f3-add1-d26c3c04dda2" />

Updated in 7f1f7f0

damian-polewski committed:
fix: update copy part2

florent-leborgne reviewed (approved):
Current copy LGTM. Let me know if you integrate more content as per @alisonelizabeth's [feedback](https://github.com/elastic/kibana/pull/233998#issuecomment-3293029321) and would like me to take a look

damian-polewski committed:
fix: hook cr changes

damian-polewski commented on x-pack/platform/plugins/shared/index_management/public/application/hooks/use_convert_index_to_lookup/use_convert_index_to_lookup.tsx:
```diff
@@ -0,0 +1,84 @@
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  const pollIntervalIdRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
```
Great idea! Updated in 8cd4eb5

damian-polewski commented on x-pack/platform/plugins/shared/index_management/public/application/hooks/use_convert_index_to_lookup/use_convert_index_to_lookup.tsx:
```diff
@@ -0,0 +1,84 @@
+    return () => {
+      clearPollInterval();
+    };
+  }, [clearPollInterval]);
```
Good catch! Updated in 8cd4eb5

damian-polewski committed:
fix: hook cr changes part2

damian-polewski committed:
fix: unit tests cr changes

kapral18 commented on x-pack/platform/plugins/shared/index_management/public/application/hooks/use_convert_index_to_lookup/use_convert_index_to_lookup.tsx:
```diff
@@ -0,0 +1,84 @@
+
+    if (error) {
+      setErrorMessage(error.message);
+      setIsConverting(false);
```
oh I meant we still show the error while retrying. But yeah whatever feels right to you. Feel free to resolve

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339882)
* Commit: ee67fe3f4db943547f2736879018fa9ea93d66ac
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-233998-ee67fe3f4db9`

### Failed CI Steps


### History
* :green_heart: [Build #339144](https://buildkite.com/elastic/kibana-pull-request/builds/339144) succeeded e758a9edb17b8fc6d8626e300e0ab68b27b39075
* :green_heart: [Build #338748](https://buildkite.com/elastic/kibana-pull-request/builds/338748) succeeded 9b92d59053a45ac16cfc410feb00324b1c2fda61
* :broken_heart: [Build #338681](https://buildkite.com/elastic/kibana-pull-request/builds/338681) failed ee923707cc398369343ecb4ac9db54ef9c9af707
* :broken_heart: [Build #338271](https://buildkite.com/elastic/kibana-pull-request/builds/338271) failed 83326dfad91464a02fb345eee1c37ea0025f29da

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339882","number":339882,"commit":"ee67fe3f4db943547f2736879018fa9ea93d66ac"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339144","number":339144,"commit":"e758a9edb17b8fc6d8626e300e0ab68b27b39075"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338748","number":338748,"commit":"9b92d59053a45ac16cfc410feb00324b1c2fda61"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338681","number":338681,"commit":"ee923707cc398369343ecb4ac9db54ef9c9af707"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338271","number":338271,"commit":"83326dfad91464a02fb345eee1c37ea0025f29da"}],"number":339882}
buildkite-pr-comment-->

kapral18 reviewed (approved):
Great job. 🚀  

Thanks for considering the feedback ❤️

damian-polewski committed:
fix: convert not available when index has 0 docs

damian-polewski committed:
fix: missing gap

damian-polewski commented:
Hi @EdLewisEL, good catch! There should be a small gap in between. I added it in 271d24f

<img width="368" height="156" alt="Screenshot 2025-09-17 at 12 55 46" src="https://github.com/user-attachments/assets/a153cb36-528e-48ad-89ba-6346f789c01b" />

jovana-andjelkovic commented:
@alisonelizabeth @damian-polewski 

1. Regarding adding description in the modal, given that both Alison and @florent-leborgne had the same question, it might be fair to say my assumption is likely wrong :D Let's add something in. 

> Converting an index to a Lookup index will create a new index with Lookup mode set. The old index will continue to exist.

Maybe something along those lines?


2. We should add a toast when user clicks "x", I agree! Great catch ❤️ 
3. Makes sense, but also not a big deal if we don't allow it as there's nothing to convert.
4. I don't think we should allow conversion of system indices, but maybe that's more of a question for @ghudgins

damian-polewski committed:
feat: disable conversion to lookup when reindex-service is not available

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340785)
* Commit: e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/340785#01995c90-35a6-4d71-8787-c65fc0d27ec9)

### History
* :yellow_heart: [Build #340304](https://buildkite.com/elastic/kibana-pull-request/builds/340304) was flaky 271d24f9815deebe0987b657ad25ec9f63a93326
* :green_heart: [Build #339882](https://buildkite.com/elastic/kibana-pull-request/builds/339882) succeeded ee67fe3f4db943547f2736879018fa9ea93d66ac
* :green_heart: [Build #339144](https://buildkite.com/elastic/kibana-pull-request/builds/339144) succeeded e758a9edb17b8fc6d8626e300e0ab68b27b39075
* :green_heart: [Build #338748](https://buildkite.com/elastic/kibana-pull-request/builds/338748) succeeded 9b92d59053a45ac16cfc410feb00324b1c2fda61
* :broken_heart: [Build #338681](https://buildkite.com/elastic/kibana-pull-request/builds/338681) failed ee923707cc398369343ecb4ac9db54ef9c9af707

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340785","number":340785,"commit":"e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340304","number":340304,"commit":"271d24f9815deebe0987b657ad25ec9f63a93326"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339882","number":339882,"commit":"ee67fe3f4db943547f2736879018fa9ea93d66ac"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339144","number":339144,"commit":"e758a9edb17b8fc6d8626e300e0ab68b27b39075"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338748","number":338748,"commit":"9b92d59053a45ac16cfc410feb00324b1c2fda61"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338681","number":338681,"commit":"ee923707cc398369343ecb4ac9db54ef9c9af707"}],"number":340785}
buildkite-pr-comment-->

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

damian-polewski committed:
fix: ts errors

damian-polewski committed:
feat: add convert modal description, add toast and update confirm deletion modal description

damian-polewski commented:
During yesterday's team sync meeting we noticed that user can close the conversion modal and delete the source index which stops the reindex operation. This is also the case when reindexing documents in Upgrade Assistant - right after the reindexing starts user can navigate to Index Management and delete the index. After discussing this with @jovana-andjelkovic we decided to update the description in the confirm deletion modal to inform the user that the index deletion also affects reindexing operation.

@florent-leborgne could you please take a look at the new copy added?

List of changes:
1. Added description to the conversion modal.
<img width="783" height="373" alt="Screenshot 2025-09-18 at 16 04 14" src="https://github.com/user-attachments/assets/e5688de1-c68d-40f9-ac1f-7b7610dae0aa" />

2. Added a toast when user closes the modal with the `x` button and the reindexing is in progress.
<img width="340" height="125" alt="Screenshot 2025-09-18 at 16 05 15" src="https://github.com/user-attachments/assets/054a7dd6-f301-4afc-85dd-008ca054c328" />

3. Added information about index deletion affecting ongoing reindex operations.
<img width="787" height="324" alt="Screenshot 2025-09-18 at 16 04 04" src="https://github.com/user-attachments/assets/3bf6673f-aacd-4ecd-b008-50528344fba5" />

cc @alisonelizabeth

florent-leborgne commented on x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/details_page/convert_to_lookup_index_modal/convert_to_lookup_index_modal.tsx:
```diff
@@ -97,16 +107,23 @@ export const ConvertToLookupIndexModal = ({ onCloseModal, onConvert, sourceIndex
       <EuiModalBody>
+        <FormattedMessage
+          id="xpack.idxMgmt.convertToLookupIndexModal.modalBodyDescription"
+          defaultMessage="Converting an index to a Lookup index will create a new index with Lookup mode set. The old index will continue to exist."
```
```suggestion
          defaultMessage="A new lookup index will be created, and the original index will still exist as before."
```
Shorter, straight to the point

florent-leborgne commented on x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/details_page/convert_to_lookup_index_modal/convert_to_lookup_index_modal_container.tsx:
```diff
@@ -0,0 +1,61 @@
+        i18n.translate(
+          'xpack.idxMgmt.convertToLookupIndexAction.indexConversionInProgressToastMessage',
+          {
+            defaultMessage: 'The conversion operation is still in progress for {sourceIndexName}.',
```
```suggestion
            defaultMessage: '{sourceIndexName} conversion into a lookup index is still in progress.',
```
doesn't add that much value, is it mandatory to have a title + body? We could combine the information into just a title or just the body.

damian-polewski replied:
Updated in 9d77bb8

florent-leborgne commented on x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/index_actions_context_menu/index_actions_context_menu.js:
```diff
@@ -419,7 +471,7 @@ export class IndexActionsContextMenu extends Component {
           <FormattedMessage
             id="xpack.idxMgmt.indexActionsMenu.deleteIndex.deleteWarningDescription"
-            defaultMessage="You can't recover a deleted index. Make sure you have appropriate backups."
+            defaultMessage="You can't recover a deleted index. Make sure you have appropriate backups. Deleting an index will affect ongoing reindex operations."
```
I'm not sure what this means. Can we be more explicit?

damian-polewski replied:
Sounds good! Updated in 41ff9a6

florent-leborgne reviewed (commented):
Hi! Took a look at the new copy 👀

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340901)
* Commit: 52612ab6cef6b407b76c1d8d8b9f0756213a81de
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-233998)

### Failed CI Steps
* [Jest Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/340901#01995d57-93f8-4c89-bf10-d2088dba024c)
* [Jest Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/340901#01995d7d-49af-4990-83af-9ac59cdec0cd)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/340901#01995d57-93f8-4c89-bf10-d2088dba024c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/340901/jobs/01995d57-93f8-4c89-bf10-d2088dba024c/artifacts/01995d7d-33fc-4517-b4e7-bd50e6df2fe8) Jest Tests #<span></span>12 / index table should show the right context menu options when one index is selected and closed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/340901#01995d7d-49af-4990-83af-9ac59cdec0cd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/340901/jobs/01995d7d-49af-4990-83af-9ac59cdec0cd/artifacts/01995d87-76f7-4d87-af15-1a4d7bc4d635) Jest Tests #<span></span>12 / index table should show the right context menu options when one index is selected and closed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/340901#01995d57-93f8-4c89-bf10-d2088dba024c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/340901/jobs/01995d57-93f8-4c89-bf10-d2088dba024c/artifacts/01995d7d-33fb-4df3-bc05-83713a2ab36f) Jest Tests #<span></span>12 / index table should show the right context menu options when one index is selected and open
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/340901#01995d7d-49af-4990-83af-9ac59cdec0cd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/340901/jobs/01995d7d-49af-4990-83af-9ac59cdec0cd/artifacts/01995d87-76f7-42fa-bf86-468ab3653657) Jest Tests #<span></span>12 / index table should show the right context menu options when one index is selected and open

### History
* :broken_heart: [Build #340813](https://buildkite.com/elastic/kibana-pull-request/builds/340813) failed 4929101e38b2b410c49ab8f39fdc2a38db4b5cd6
* :broken_heart: [Build #340785](https://buildkite.com/elastic/kibana-pull-request/builds/340785) failed e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b
* :yellow_heart: [Build #340304](https://buildkite.com/elastic/kibana-pull-request/builds/340304) was flaky 271d24f9815deebe0987b657ad25ec9f63a93326
* :green_heart: [Build #339882](https://buildkite.com/elastic/kibana-pull-request/builds/339882) succeeded ee67fe3f4db943547f2736879018fa9ea93d66ac
* :green_heart: [Build #339144](https://buildkite.com/elastic/kibana-pull-request/builds/339144) succeeded e758a9edb17b8fc6d8626e300e0ab68b27b39075
* :green_heart: [Build #338748](https://buildkite.com/elastic/kibana-pull-request/builds/338748) succeeded 9b92d59053a45ac16cfc410feb00324b1c2fda61

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340901","number":340901,"commit":"52612ab6cef6b407b76c1d8d8b9f0756213a81de"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340813","number":340813,"commit":"4929101e38b2b410c49ab8f39fdc2a38db4b5cd6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340785","number":340785,"commit":"e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340304","number":340304,"commit":"271d24f9815deebe0987b657ad25ec9f63a93326"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339882","number":339882,"commit":"ee67fe3f4db943547f2736879018fa9ea93d66ac"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339144","number":339144,"commit":"e758a9edb17b8fc6d8626e300e0ab68b27b39075"}],"number":340901}
buildkite-pr-comment-->

damian-polewski committed:
fix: update copy

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/341376)
* Commit: 9d77bb80b399e001d51afc10dd47040a70c6fd84
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-233998)

### Failed CI Steps
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/341376#019961de-633a-4012-a9f5-3888bf7267f1)
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/341376#01996202-eb16-41d1-ac7b-13c781da9119)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/341376#019961de-633a-4012-a9f5-3888bf7267f1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/341376/jobs/019961de-633a-4012-a9f5-3888bf7267f1/artifacts/01996202-c8e8-4cc4-a572-5876c5316083) Jest Tests #<span></span>13 / index table should show the right context menu options when one index is selected and closed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/341376#01996202-eb16-41d1-ac7b-13c781da9119) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/341376/jobs/01996202-eb16-41d1-ac7b-13c781da9119/artifacts/0199620d-b7df-4fb2-9df3-21cddd3ba338) Jest Tests #<span></span>13 / index table should show the right context menu options when one index is selected and closed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/341376#019961de-633a-4012-a9f5-3888bf7267f1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/341376/jobs/019961de-633a-4012-a9f5-3888bf7267f1/artifacts/01996202-c8e8-4687-9d75-89dca7de1a4f) Jest Tests #<span></span>13 / index table should show the right context menu options when one index is selected and open
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/341376#01996202-eb16-41d1-ac7b-13c781da9119) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/341376/jobs/01996202-eb16-41d1-ac7b-13c781da9119/artifacts/0199620d-b7df-475b-8ed7-f2991191eddd) Jest Tests #<span></span>13 / index table should show the right context menu options when one index is selected and open

### History
* :broken_heart: [Build #340901](https://buildkite.com/elastic/kibana-pull-request/builds/340901) failed 52612ab6cef6b407b76c1d8d8b9f0756213a81de
* :broken_heart: [Build #340813](https://buildkite.com/elastic/kibana-pull-request/builds/340813) failed 4929101e38b2b410c49ab8f39fdc2a38db4b5cd6
* :broken_heart: [Build #340785](https://buildkite.com/elastic/kibana-pull-request/builds/340785) failed e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b
* :yellow_heart: [Build #340304](https://buildkite.com/elastic/kibana-pull-request/builds/340304) was flaky 271d24f9815deebe0987b657ad25ec9f63a93326
* :green_heart: [Build #339882](https://buildkite.com/elastic/kibana-pull-request/builds/339882) succeeded ee67fe3f4db943547f2736879018fa9ea93d66ac

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341376","number":341376,"commit":"9d77bb80b399e001d51afc10dd47040a70c6fd84"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340901","number":340901,"commit":"52612ab6cef6b407b76c1d8d8b9f0756213a81de"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340813","number":340813,"commit":"4929101e38b2b410c49ab8f39fdc2a38db4b5cd6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340785","number":340785,"commit":"e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340304","number":340304,"commit":"271d24f9815deebe0987b657ad25ec9f63a93326"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339882","number":339882,"commit":"ee67fe3f4db943547f2736879018fa9ea93d66ac"}],"number":341376}
buildkite-pr-comment-->

damian-polewski committed:
fix: update copy part 2

damian-polewski committed:
fix: unit test

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/341480)
* Commit: c557dc0a20a311f9aba2038419ea0d29da4c2a2c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-233998)

### Failed CI Steps
* [FTR Configs #112](https://buildkite.com/elastic/kibana-pull-request/builds/341480#01996263-f123-4c0e-b9c1-9f52053408d1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/341480#01996263-f123-4c0e-b9c1-9f52053408d1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/341480/jobs/01996263-f123-4c0e-b9c1-9f52053408d1/artifacts/019962a1-b5a1-4345-8e3b-f25e71fe285b) FTR Configs #<span></span>112 / discover/security/context_awareness cell renderer ES|QL mode should render alert workflow status badge

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/b18c9c547f94a7e3b2b4d6b8349e4d6614e3e783) | [after](https://github.com/elastic/kibana/commit/c557dc0a20a311f9aba2038419ea0d29da4c2a2c) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 673 | 680 | +7 |
| `upgradeAssistant` | 181 | 182 | +1 |
| total |  |  | +8 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/b18c9c547f94a7e3b2b4d6b8349e4d6614e3e783) | [after](https://github.com/elastic/kibana/commit/c557dc0a20a311f9aba2038419ea0d29da4c2a2c) | diff |
| --- | --- | --- | --- |
| `@kbn/index-management-shared-types` | 149 | 150 | +1 |
| `@kbn/upgrade-assistant-pkg-common` | 10 | 11 | +1 |
| `indexManagement` | 250 | 253 | +3 |
| `reindexService` | 55 | 54 | -1 |
| total |  |  | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b18c9c547f94a7e3b2b4d6b8349e4d6614e3e783) | [after](https://github.com/elastic/kibana/commit/c557dc0a20a311f9aba2038419ea0d29da4c2a2c) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 689.5KB | 696.2KB | +6.7KB |
| `upgradeAssistant` | 228.7KB | 228.1KB | -569.0B |
| total |  |  | +6.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b18c9c547f94a7e3b2b4d6b8349e4d6614e3e783) | [after](https://github.com/elastic/kibana/commit/c557dc0a20a311f9aba2038419ea0d29da4c2a2c) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 31.8KB | 32.0KB | +113.0B |
| `reindexService` | 3.3KB | 3.1KB | -216.0B |
| total |  |  | -103.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/b18c9c547f94a7e3b2b4d6b8349e4d6614e3e783) | [after](https://github.com/elastic/kibana/commit/c557dc0a20a311f9aba2038419ea0d29da4c2a2c) | diff |
| --- | --- | --- | --- |
| `@kbn/index-management-shared-types` | 149 | 150 | +1 |
| `@kbn/upgrade-assistant-pkg-common` | 10 | 11 | +1 |
| `indexManagement` | 255 | 258 | +3 |
| `reindexService` | 62 | 61 | -1 |
| total |  |  | +4 |


</details>


### History
* :broken_heart: [Build #341376](https://buildkite.com/elastic/kibana-pull-request/builds/341376) failed 9d77bb80b399e001d51afc10dd47040a70c6fd84
* :broken_heart: [Build #340901](https://buildkite.com/elastic/kibana-pull-request/builds/340901) failed 52612ab6cef6b407b76c1d8d8b9f0756213a81de
* :broken_heart: [Build #340813](https://buildkite.com/elastic/kibana-pull-request/builds/340813) failed 4929101e38b2b410c49ab8f39fdc2a38db4b5cd6
* :broken_heart: [Build #340785](https://buildkite.com/elastic/kibana-pull-request/builds/340785) failed e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b
* :yellow_heart: [Build #340304](https://buildkite.com/elastic/kibana-pull-request/builds/340304) was flaky 271d24f9815deebe0987b657ad25ec9f63a93326

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341480","number":341480,"commit":"c557dc0a20a311f9aba2038419ea0d29da4c2a2c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341376","number":341376,"commit":"9d77bb80b399e001d51afc10dd47040a70c6fd84"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340901","number":340901,"commit":"52612ab6cef6b407b76c1d8d8b9f0756213a81de"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340813","number":340813,"commit":"4929101e38b2b410c49ab8f39fdc2a38db4b5cd6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340785","number":340785,"commit":"e5a5605d0eec0dc5af5201a2fd30e2d2f27dea7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340304","number":340304,"commit":"271d24f9815deebe0987b657ad25ec9f63a93326"}],"number":341480}
buildkite-pr-comment-->

mattkime committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

mattkime commented on x-pack/platform/plugins/shared/index_management/public/application/hooks/use_convert_index_to_lookup/use_convert_index_to_lookup.tsx:
```diff
@@ -0,0 +1,108 @@
+          i18n.translate(
+            'xpack.idxMgmt.convertToLookupIndexAction.reindexFailedOrCancelledErrorMessage',
+            {
+              defaultMessage: 'Reindex did not complete successfully.',
```
Might be good to say `Reindex did not complete successfully due to unknown error.` since we do return the message if its provided

mattkime commented on x-pack/platform/plugins/shared/index_management/public/application/services/api.ts:
```diff
@@ -482,3 +488,21 @@ export function useLoadInferenceEndpoints() {
+export const startReindex = (sourceIndexName: string, lookupIndexName: string) => {
+  return reindexService.startReindex({
+    indexName: sourceIndexName,
+    newIndexName: lookupIndexName,
```
I think it would be a good idea to delete the old index and provide an alias from old to new. This is available via one of the startReindex params

mattkime commented:
Previously, the reindexing service always created a unique pattern for new indices but now anything can be provided. It should make a check using the `resolve` api to see if the new index name matches an existing index.

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

damian-polewski committed:
fix: update copy part 3

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/341969)
* Commit: f0dd1daf55a89a37ccf7e7232d709c0177a513ce
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-233998)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/fa205bc267e79473defa98cee0af991ee8ca7ba8) | [after](https://github.com/elastic/kibana/commit/f0dd1daf55a89a37ccf7e7232d709c0177a513ce) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 665 | 672 | +7 |
| `upgradeAssistant` | 181 | 182 | +1 |
| total |  |  | +8 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/fa205bc267e79473defa98cee0af991ee8ca7ba8) | [after](https://github.com/elastic/kibana/commit/f0dd1daf55a89a37ccf7e7232d709c0177a513ce) | diff |
| --- | --- | --- | --- |
| `@kbn/index-management-shared-types` | 149 | 150 | +1 |
| `@kbn/upgrade-assistant-pkg-common` | 10 | 11 | +1 |
| `indexManagement` | 250 | 253 | +3 |
| `reindexService` | 55 | 54 | -1 |
| total |  |  | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/fa205bc267e79473defa98cee0af991ee8ca7ba8) | [after](https://github.com/elastic/kibana/commit/f0dd1daf55a89a37ccf7e7232d709c0177a513ce) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 688.6KB | 695.3KB | +6.7KB |
| `upgradeAssistant` | 228.7KB | 228.1KB | -569.0B |
| total |  |  | +6.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/fa205bc267e79473defa98cee0af991ee8ca7ba8) | [after](https://github.com/elastic/kibana/commit/f0dd1daf55a89a37ccf7e7232d709c0177a513ce) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 31.8KB | 32.0KB | +113.0B |
| `reindexService` | 3.3KB | 3.1KB | -216.0B |
| total |  |  | -103.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/fa205bc267e79473defa98cee0af991ee8ca7ba8) | [after](https://github.com/elastic/kibana/commit/f0dd1daf55a89a37ccf7e7232d709c0177a513ce) | diff |
| --- | --- | --- | --- |
| `@kbn/index-management-shared-types` | 149 | 150 | +1 |
| `@kbn/upgrade-assistant-pkg-common` | 10 | 11 | +1 |
| `indexManagement` | 255 | 258 | +3 |
| `reindexService` | 62 | 61 | -1 |
| total |  |  | +4 |


</details>


### History
* :broken_heart: [Build #341817](https://buildkite.com/elastic/kibana-pull-request/builds/341817) failed 58fec71964607e3122d9ca1107fc8a51d70067b4
* :broken_heart: [Build #341777](https://buildkite.com/elastic/kibana-pull-request/builds/341777) failed 17e7b16d458884741bfb29b41bf8b0806b8e2610
* :yellow_heart: [Build #341480](https://buildkite.com/elastic/kibana-pull-request/builds/341480) was flaky c557dc0a20a311f9aba2038419ea0d29da4c2a2c
* :broken_heart: [Build #341376](https://buildkite.com/elastic/kibana-pull-request/builds/341376) failed 9d77bb80b399e001d51afc10dd47040a70c6fd84
* :broken_heart: [Build #340901](https://buildkite.com/elastic/kibana-pull-request/builds/340901) failed 52612ab6cef6b407b76c1d8d8b9f0756213a81de

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341969","number":341969,"commit":"f0dd1daf55a89a37ccf7e7232d709c0177a513ce"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341817","number":341817,"commit":"58fec71964607e3122d9ca1107fc8a51d70067b4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341777","number":341777,"commit":"17e7b16d458884741bfb29b41bf8b0806b8e2610"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341480","number":341480,"commit":"c557dc0a20a311f9aba2038419ea0d29da4c2a2c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341376","number":341376,"commit":"9d77bb80b399e001d51afc10dd47040a70c6fd84"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340901","number":340901,"commit":"52612ab6cef6b407b76c1d8d8b9f0756213a81de"}],"number":341969}
buildkite-pr-comment-->

damian-polewski committed:
Merge branch 'main' into index_management/add_option_to_convert_to_lookup_in_manage_dropdown

damian-polewski committed:
fix: dont allow converting hidden indices

damian-polewski committed:
fix: handle unexpected throw errors
